File: 234-Palindrome-Linked-List/234.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def isPalindrome(self, head: ListNode) -> bool:
    arr = []
    curr = head
    while curr != None:
        arr.append(curr.val)
        curr = curr.next
    return arr == arr[::-1]

def main():
    pass

if __name__ == "__main__":
    pass

File: 234-Palindrome-Linked-List/234.cpp
#include <iostream>

#include <vector>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(nullptr) {};
};

bool isPalindrome(ListNode* head) {
    ListNode* curr = head;
    
    vector<int> arr = vector<int>();
    while (curr != nullptr) {
        arr.push_back(curr->val);
        curr = curr->next;
    }

    int front, back;
    front = 0;
    back = arr.size() - 1;
    while(front < back) {
        if(arr[front] != arr[back])
            return false;
        front++;
        back--;
    }

    return true;
}

int main(){
    return 0;
}


